### PR TITLE
feat: add luis

### PR DESCRIPTION
**Pull request by z4o4z**

### CODE REVIEW

**Review by DecathectZero — APPROVED** [review]

File: lib/controllers/interact.ts [hunk 1]
```diff
@@ -1,28 +1,42 @@
...
     const {
       body: { state, request },
       params: { versionID },
     } = req;
 
+    const { projectID } = await dataAPI.getVersion(versionID);
+    const { prototype } = await dataAPI.getProject(projectID);
+
+    if (!prototype?.nlp) {
+      throw new VError('Prototype is not rendered!');
```
[the remark this answers is not in the document]
Comment (z4o4z) [reply lib/controllers/interact.ts:28]: for now, NLP filled during the render, but yeah, will update the message

File: lib/services/nlu/index.ts [hunk 2]
```diff
@@ -7,7 +7,19 @@ export const utils = {};
...
+  handle = async (context: Context) => {
+    const { nlp, query } = context.request;
+
+    if (!query) {
+      return context;
+    }
+
+    const data = await this.services.luis.predict({ query, appID: nlp!.appID });
+
+    console.log(data);
```
[the remark this answers is not in the document]
Comment (frankgu968) [reply lib/services/nlu/index.ts:19]: Actually, we should attach the returned data into the context as `context.request.nlpResult`? Then the downstream services can consume this

File: lib/controllers/interact.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (DecathectZero) [reply lib/controllers/interact.ts:28]: we can just throw this error inside the nlu handler

File: lib/services/nlu/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (DecathectZero) [reply lib/services/nlu/index.ts:19]: yeah I think evgeny only added this for testing purposes, this isn't complete until I have all the request types defined

**Comment by DecathectZero** [comment]
planning on this instead:
https://github.com/voiceflow/general-runtime/pull/10

**Pull request closed without merging**